src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
The only thing I was wondering about this (and TransactionEvent) is whether it might make sense to make TransactionEvent have two fields. One is the transaction object and the second is the status. I would go further as to say for now we don't actually provide the transaction object so it would be more about future proofing the stateChange operation.

The API would then be:
 void stateChange(TransactionEvent transactionEvent);

The reason you don't need the transaction object is that Scott for example (who is our use case) won't use it as he is setting a unique listener per transaction. It would only be the person that set the same listener instance for different transactions and if TransactionEvent could in future be modified to pass that reference back (so not make it a struct now) then it would not be such a big deal as someone can go on our forum "how do I get the transaction", "oh we can add that as JBTM-xyz"

We don't know whether someone will ever need the transaction and the only user we do have isn't asking for it. Another reason to make TransactionEvent "future-extensible" is if someone wants us to pass back various other bits of meta data (the object store used, the thread that created a transaction) whatever it might be we can add them in in future.

Its more like JMS MessageListener then which onMessage just takes a Message and all the data is linked from that.

What do you think?